ext/nio4r/org/nio4r/Selector.java

```diff
@@ -136,7 +137,10 @@ public IRubyObject register(ThreadContext context, IRubyObject io, IRubyObject i
     @JRubyMethod
     public IRubyObject deregister(ThreadContext context, IRubyObject io) {
         Ruby runtime = context.getRuntime();
-        Channel rawChannel = RubyIO.convertToIO(context, io).getChannel();
+        OpenFile f = RubyIO.convertToIO(context, io).getOpenFileInitialized();
+        if (f.fd() == null)
+            return context.nil;
```

Comment:
I'm wondering when this might occur and if it might be worth at least logging it happened or something.